packages/vite/src/node/utils.ts

```diff
@@ -534,6 +534,38 @@ export function removeDirSync(dir: string) {
   }
 }
 
+const REMOVE_DIR_TIMEOUT = 5000
+const rmdir = promisify(fs.rm ?? fs.rmdir) // TODO: Remove after support for Node 12 is dropped
```

Comment:
haven't we already dropped support for Node 12?

Reply:
cleaning up the others here: https://github.com/vitejs/vite/pull/8239

Reply:
Think we would still leave this in if this fix were to be backported to Vite 2 🤔

Reply:
I figured I didn't want to make those kind of changes as part of this PR. 
If all Node 12 related TODOs are removed on main before this PR lands then I can remove it as part of a rebase, otherwise I think they should stay